src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -163,6 +163,10 @@ public function setParameter($name, $value)
      */
     public function set($id, $service)
     {
+        if (preg_match('/[^a-z0-9\._%]/i', $id)) {
+            @trigger_error(sprintf('Using characters other than A-Z, 0-9, ., and _ in service ids ("%s" given) is deprecated since Symfony 3.1 and will throw an exception in 4.0.', $id), E_USER_DEPRECATED);
```

Comment:
Instead of:

```
... and will throw an exception in 4.0.
```

We use this other wording in most deprecation notices:

```
... and will be removed in 4.0.
```